ksql-engine/src/main/java/io/confluent/ksql/codegen/SqlToJavaVisitor.java

```diff
@@ -414,47 +414,52 @@ private String getCastToBooleanString(Schema schema, String exprStr) {
     }
 
     private String getCastToIntegerString(Schema schema, String exprStr) {
-      if (schema == Schema.STRING_SCHEMA) {
-        return "Integer.parseInt(" + exprStr + ")";
-      } else if (schema == Schema.INT32_SCHEMA) {
-        return exprStr;
-      } else if (schema == Schema.INT64_SCHEMA) {
-        return "(new Long(" + exprStr + ").intValue())";
-      } else if (schema == Schema.FLOAT64_SCHEMA) {
-        return "(new Double(" + exprStr + ").intValue())";
-      } else {
-        throw new KsqlFunctionException(
-            "Invalid cast operation: Cannot cast " + exprStr + " to Integer.");
+      switch (schema.type()) {
```

Comment:
Would it be possible to push down this logic into Schema.TYPE enums? - i.e. implement castToIntegerString() on each enum? You also get comparitors and a bunch of other stuff for free.